server/server/src/file_service.rs

```diff
@@ -804,15 +804,59 @@ pub async fn admin_rebuild_index(
         .index_db
         .transaction(|tx| match context.request.index {
             ServerIndex::OwnedFiles => {
+                let mut owned_files = HashMap::new();
+                for owner in tx.accounts.keys() {
+                    owned_files.insert(*owner, HashSet::new());
+                }
+                for (id, file) in tx.metas.get_all() {
+                    if let Some(owned_files) = owned_files.get_mut(&file.owner()) {
+                        owned_files.insert(*id);
+                    }
+                }
+
                 tx.owned_files.clear();
+                for (k, v) in owned_files {
+                    tx.owned_files.insert(k, v);
+                }
+                Ok(())
+            }
+            ServerIndex::SharedFiles => {
+                let mut shared_files = HashMap::new();
+                for owner in tx.accounts.keys() {
+                    shared_files.insert(*owner, HashSet::new());
+                }
                 for (id, file) in tx.metas.get_all() {
-                    let mut set = tx
-                        .owned_files
-                        .get(&file.owner())
-                        .cloned()
-                        .unwrap_or_default();
-                    set.insert(*id);
-                    tx.owned_files.insert(file.owner(), set);
```

Comment:
iiuc this doesn't create empty sets for users with no files, which is prob fine bc each user has a root, but I did it more explicitly for clarity and consistency with how other indexes are rebuilt